barretenberg/cpp/src/barretenberg/flavor/generated/Fib_flavor.hpp

```diff
@@ -0,0 +1,322 @@
+
+
+#pragma once
+#include "../relation_definitions_fwd.hpp"
+#include "barretenberg/commitment_schemes/kzg/kzg.hpp"
+#include "barretenberg/ecc/curves/bn254/g1.hpp"
+#include "barretenberg/polynomials/barycentric.hpp"
+#include "barretenberg/polynomials/univariate.hpp"
+
+#include "barretenberg/flavor/flavor.hpp"
+#include "barretenberg/polynomials/evaluation_domain.hpp"
+#include "barretenberg/polynomials/polynomial.hpp"
+#include "barretenberg/relations/generated/Fib.hpp"
+#include "barretenberg/transcript/transcript.hpp"
+
+namespace proof_system::honk {
+namespace flavor {
+
+class FibFlavor {
+  public:
+    using Curve = curve::BN254;
+    using G1 = Curve::Group;
+    using PCS = pcs::kzg::KZG<Curve>;
+
+    using FF = G1::subgroup_field;
+    using Polynomial = barretenberg::Polynomial<FF>;
+    using PolynomialHandle = std::span<FF>;
+    using GroupElement = G1::element;
+    using Commitment = G1::affine_element;
+    using CommitmentHandle = G1::affine_element;
+    using CommitmentKey = pcs::CommitmentKey<Curve>;
+    using VerifierCommitmentKey = pcs::VerifierCommitmentKey<Curve>;
+
+    static constexpr size_t NUM_PRECOMPUTED_ENTITIES = 2;
+    static constexpr size_t NUM_WITNESS_ENTITIES = 2;
+    static constexpr size_t NUM_WIRES = NUM_WITNESS_ENTITIES + NUM_PRECOMPUTED_ENTITIES;
```

Comment:
NUM_WIRES usually means the number of witness used by the circuit builder (so these are the witnesses that have just the arithmetic elements, without witness polynomials which help in creating the proof, such as grand product polynomials, inverse polynomials, etc)